lib/handler/connect.c

```diff
@@ -52,21 +52,137 @@ struct st_connect_request_t {
         size_t next;
     } server_addresses;
     h2o_timer_t timeout;
+    struct {
+        h2o_iovec_t dest_hostname;
+        const struct st_server_address_t *dest_addr;
```

Comment:
I think we can omit this.

We can consult `st_connect_request_t::server_addresses.size != 0` to determine if name resolution is in progress. The last address to which we tried to connect can be found at `server_addresses.list[server_addresses_next - 1]`.

We might want to make small tweaks like changing the code to retain "current" index rather than "next," but I doubt if having a two ways to refer to the address being tested is worthwhile.

Reply:
It seemed a little bit "magical" to me to be doing `server_addresses.list[server_addresses.next - 1]`, so I denormalized the information.  I like the suggestion to have a "current" index.  `server_addresses.list[server_addresses.current]` would work, so I will see if I can do that instead, if you agree.  Thank you!

Reply:
ad64214be removes the code that tries to populate `next-hop`, which simplifies things a lot.